homeassistant/components/ecobee/climate.py

```diff
@@ -536,3 +605,54 @@ def hold_preference(self):
         # supported; note that this should not include 'indefinite'
         # as an indefinite away hold is interpreted as away_mode
         return "nextTransition"
+
+    def create_vacation(self, service_data):
+        """Create a vacation with user-specified parameters."""
+        vacation_name = service_data[ATTR_VACATION_NAME]
+        cool_temp = service_data[ATTR_COOL_TEMP]
+        heat_temp = service_data[ATTR_HEAT_TEMP]
+
+        start_date = service_data.get(ATTR_START_DATE)
+        start_time = service_data.get(ATTR_START_TIME)
+        end_date = service_data.get(ATTR_END_DATE)
+        end_time = service_data.get(ATTR_END_TIME)
+        fan_mode = service_data[ATTR_FAN_MODE]
+        fan_min_on_time = service_data[ATTR_FAN_MIN_ON_TIME]
+
+        if cool_temp < 35:
+            # Assume user has specified cool_temp and heat_temp in celsius
+            # Convert to fahrenheit before calling function
+            cool_temp = (cool_temp * (9 / 5)) + 32
```

Comment:
We have a convert temperature util:
https://github.com/home-assistant/home-assistant/blob/770ad86f126eb890f6c4021dc72261af62acd862/homeassistant/util/temperature.py#L24-L26